### PR TITLE
writer: introducing new mention feature

### PR DESCRIPTION
**Pull request by Rash419**
Signed-off-by: Rash419 <rashesh.padia@collabora.com>
Change-Id: I326cd663737a2e8e04bedeccd8f130fec4014c26

* Core patch: https://gerrit.libreoffice.org/c/core/+/140015
* Resolves: # 
* Target version: master

### CODE REVIEW

**Comment by Rash419** [comment]
@eszkadev @pedropintosilva @kendy  to test this just open framed.doc.html start typing mention
review, feedback, suggestion appreciated

File: browser/src/control/Control.JSDialogBuilder.js [hunk 1]
```diff
@@ -1920,9 +1920,34 @@ L.Control.JSDialogBuilder = L.Control.extend({
...
+					else
+						listElements.eq(currIndex + 1).find(spanElement).focus();
+					preventDef = true;
+				} else if (event.key === 'ArrowUp') {
+					if (currIndex === 0)
+						listElements.eq(treeLength - 1).find(spanElement).focus();
+					else
+						listElements.eq(currIndex - 1).find(spanElement).focus();
+					preventDef = true;
+				} else if (treeViewData.id === 'mentionList' && event.key !== 'Tab' && event.key !== 'Shift') {
```
Comment (eszkadev) [inline browser/src/control/Control.JSDialogBuilder.js:1944]: hmm, I would prefer to keep widgets generic, not having any logic for particular widget instances for custom dialogs.
Can we do it different way?

For example pass custom keydown callback to the widget which will be called before original one and will return boolean which will indicate if event was consumed and doesn't have to be handled anymore.

**Review by pedropintosilva — COMMENTED** [review]
Thanks @Rash419 I have encountered some interesting behaviour. If zoom in (brwoser zoom) the list shows up in the wrong place:

image

**Comment by Rash419** [comment]
> Thanks @Rash419 I have encountered some interesting behaviour. If zoom in (brwoser zoom) the list shows up in the wrong place:
> 
> image

@pedropintosilva I guess with every jsdialog popup we have 
@eszkadev thougths ?

File: browser/src/control/Control.JSDialogBuilder.js (hunk 1 above)
Comment (Rash419) [reply browser/src/control/Control.JSDialogBuilder.js:1944]: yep makes sense , thanks : )

**Review by eszkadev — APPROVED** [review]
From code reading perspective looks ok now :)

**Review by eszkadev — CHANGES_REQUESTED** [review]
tested again rebased:
- nice, border is fixed now
but what I noticed:
1. type "@a"
2. select something from the list (here Alexandra)
3. you got in the document: "@a@Alexandra"

**Review by eszkadev — APPROVED** [review]
ok, needed core patch :)

**Review by pedropintosilva — APPROVED** [review]

**Pull request merged**